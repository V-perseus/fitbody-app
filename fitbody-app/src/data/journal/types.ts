import { AsyncThunkPayloadCreator } from '@reduxjs/toolkit'
import { MoodsIconMapType } from '../../config/svgs/dynamic/moodsMap'
import { ActivitiesIconMapType } from '../../config/svgs/dynamic/activitiesMap'

export const GET_MOODS = 'journal/getMoods'
export const GET_ACTIVITIES = 'journal/getActivities'
export const CREATE_JOURNAL = 'journal/createJournal'
export const UPDATE_JOURNAL = 'journal/updateJournal'
export const GET_JOURNALS = 'journal/getJournals'

export interface IJournal {
  id?: number
  user_id: number
  accomplished_today: string
  accomplish_tmrw: string
  created_at?: string
  updated_at?: string
  proud_of_today: string
  additional_thoughts: string
  entry_date: string
  main_mood_id: number | null
  mood_items: { id: number; order: number }[]
  mood_ids: number[]
  journal_activity_ids: number[]
}

export interface IMood {
  id?: number
  name: string
  icon_key: MoodsIconMapType
  created_at?: string
  updated_at?: string
}

export interface IActivity {
  id?: number
  name: string
  icon_key: ActivitiesIconMapType
  created_at?: string
  updated_at?: string
}

export interface IGetJournalsPayload {
  start: string
  end: string
}

export interface ICreateJournalPayload {
  data: any
  onSuccess: () => void
}

export interface IUpdateJournalPayload extends AsyncThunkPayloadCreator<any, IJournal> {
  journalId: string
  data: any
  onSuccess: () => void
}
