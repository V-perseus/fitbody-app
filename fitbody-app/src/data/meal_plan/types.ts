import { EatingPreferencesIconMapType } from '../../config/svgs/dynamic/eatingPreferencesMap'

export const GET_RECIPES_FOR_DAY = 'meal/getRecipesForTheDay'
export const GET_RECIPES = 'meal/getRecipes'
export const LIST_MEAL_SETS = 'meal/listMealSets'
export const GET_RECIPES_RANGE = 'meal/getRecipesRange'
export const ADD_MEALSET_TO_FAVORITES = 'meal/addMealSetToFavorites'
export const REMOVE_MEALSET_FROM_FAVORITES = 'meal/removeMealSetFromFavorites'

/* ************** ENUMS ************** */
export enum EatingPreference {
  'REGULAR' = 'REGULAR',
  'VEGETARIAN' = 'VEGETARIAN',
  'VEGAN' = 'VEGAN',
  'DAIRY-FREE' = 'DAIRY-FREE',
  'PESCATARIAN' = 'PESCATARIAN',
  'GLUTEN-FREE' = 'GLUTEN-FREE',
  'GLUTEN-FREE + DIARY-FREE' = 'GLUTEN-FREE + DIARY-FREE',
  'MEDITERRANEAN' = 'MEDITERRANEAN',
  'KETO' = 'KETO',
}
export enum RecipeType {
  'recipe' = 'recipe',
  'easy_add_recipe' = 'easy_add_recipe',
  'ingredient' = 'ingredient',
}
export enum MealTimeSlot {
  'Breakfast' = 'Breakfast',
  'Lunch' = 'Lunch',
  'Dinner' = 'Dinner',
  'Snack' = 'Snack',
  // 'Snack 1' = 'Snack 1',
  // 'Snack 2' = 'Snack 2',
  // 'Snack 3' = 'Snack 3',
}

/* ************** TYPES ************** */

export type AltMeasure = {
  serving_weight: number
  measure: string
  seq: number
  qty: number
}

export type Measure = {
  amount?: number | null
  serving_unit?: string | null
  serving_weight?: number | null
}

export interface IMealPlanType {
  id: number
  name: Capitalize<Lowercase<EatingPreference>>
  key: EatingPreference
  created_at: string
  updated_at: string
  active: boolean
  order: number
  coming_soon: boolean
  is_restrictive: 0 | 1
}

export interface IRecipeStep {
  id: number
  header: string | null
  order: number
  text?: string
  items: {
    id: number
    text: string
  }[]
}

export interface IRecipeTool {
  id: number
  name: string
}

export interface IIndividualRecipeDetails
  extends Omit<Partial<IRecipe>, 'id' | 'name' | 'type' | 'calories' | 'fats' | 'carbs' | 'protein'> {
  id: number | string
  name: string
  type: RecipeType
  calories: number
  fats: number
  carbs: number
  protein: number
  img_url: string
  thumb_img_url: string
  tags: EatingPreference[]
  meal_time_slots: MealTimeSlot[]
  total_time: number
  prep_time: number
  servings: number
  is_favorite: boolean
  user_generated: boolean
  children_ids: number[]
  instructions: string
}

// this represents a recipe created by us or as part of a meal set (not from Nix)
// a FitBody recipe
export interface IBaseRecipe extends Omit<Partial<IRecipe>, 'id' | 'name' | 'type' | 'calories' | 'fats' | 'carbs' | 'protein'> {
  type: RecipeType.recipe
  id: number
  base_id?: number
  base_recipe_id?: number
  name: string
  img_url?: string
  thumb_img_url: string
  tags: EatingPreference[]
  diary_meal_time_slot_id: number
  calories: number
  fats: number
  carbs: number
  protein: number
  user_generated?: boolean
  is_favorite?: boolean
  key?: string
  original_recipe_id?: number
}

export interface IBaseIngredient {
  type: RecipeType.ingredient
  id: number
  base_id?: number
  base_ingredient_id: number
  nix_item_id: string
  is_branded: boolean
  img_url: string
  name: string
  brand_name: string | null
  calories: number
  carbs: number
  protein: number
  fats: number
  quantity: string
  serving_unit: string
  key?: string
  diary_meal_time_slot_id?: number
}

export interface IMealPlanIngredient {
  id: number
  name: string
  brand_name: string | null
  calories: number
  protein: number
  carbs: number
  fats: number
  serving_unit: string | null
  img_url: string | null
  serving_qty: number | null
  cholesterol: number
  dietary_fiber: number
  potassium: number
  saturated_fat: number
  sodium: number
  sugars: number
  serving_weight: number
  alt_measurements: AltMeasure[] | null
  quantity: string
  nix_item_id: string | null
  nix_food_name: string | null
  meal_time_slot: number
  meal_time_slot_id: number
  meal_ingradient_id: number
  key?: string
  diary_meal_time_slot_id?: number
}

// An easy add is an ingredient and recipe
export interface IEasyAddRecipeIngredient {
  type: RecipeType.easy_add_recipe
  id: string
  base_easy_add_recipe_id?: number
  base_id?: number
  name: string
  fats: number
  carbs: number
  protein: number
  calories: number
  is_favorite?: boolean
  diary_meal_time_slot_id?: number
  quantity?: number
  key?: string
  disableRightSwipe?: boolean
}

export interface IEasyAdd {
  id?: number
  name: string
  fats: number
  carbs: number
  protein: number
  calories: number
  is_favorite: boolean
}

export interface ISingleEditIngredient {
  id: number
  calories: number
  carbs: number
  protein: number
  fats: number
  quantity: number
  serving_unit: string
  meal_time_slot_id: number
  base_ingredient_id?: number
}

/* **************** MEAL PLANS their RECIPES and INGREDIENTS ***********************/

export interface IMealPlanRecipe {
  id: number
  name: string
  tags: EatingPreferencesIconMapType[]
  // tags: EatingPreference[]
  thumb_img_url: string
  calories: number
  fats: number
  carbs: number
  protein: number
  meal_time_slots: MealTimeSlot[]
  user_generated?: boolean
  diary_meal_time_slot_id?: number
}

export interface IMealPlan {
  id: number
  name: string
  total_calories: number
  total_fats: number
  total_carbs: number
  total_protein: number
  is_favorite: boolean
  eating_preferences: EatingPreference[]
  recipes: IMealPlanRecipe[]
}

export interface IMealSet {
  id: number
  name: string
  total_calories: number
  total_fats: number
  total_carbs: number
  total_protein: number
  is_favorite: boolean
  eating_preferences: EatingPreference[]
}

export interface IMacro {
  calories: number
  fats: number
  carbs: number
  protein: number
}

export interface IMealPlanItem
  extends Omit<
    Partial<IRecipe>,
    | 'id'
    | 'name'
    | 'type'
    | 'meal_time_slot_id'
    | 'original_calories'
    | 'original_fats'
    | 'original_carbs'
    | 'original_protein'
    | 'calories'
    | 'fats'
    | 'carbs'
    | 'protein'
  > {
  id: number
  name: string
  type: RecipeType
  meal_time_slot_id: number
  original_calories: number
  original_fats: number
  original_carbs: number
  original_protein: number
  calories: number
  fats: number
  carbs: number
  protein: number
}
// part of mealPlans state
export interface IMealPlanDay {
  id: number
  date: Date
  meal_set?: IMealSet
  items: IMealPlanItem[]
}

// TODO this is going to be a problem
// Descriminated union of recipe types based on the type field
export type Recipe = IBaseRecipe | IEasyAddRecipeIngredient | IBaseIngredient //| IIndividualRecipeDetails

export type GroceryIngredient = IBaseIngredient | IEasyAddRecipeIngredient | IMealPlanIngredient

export interface IRecipe {
  base_ingredient_id?: number
  is_branded?: boolean

  base_id?: number
  base_recipe_id?: number
  diary_meal_time_slot_id?: number
  is_favorite?: boolean
  key?: string
  original_recipe_id?: number
  nix_food_name?: string

  base_easy_add_recipe_id?: number
  quantity?: number | string
  disableRightSwipe?: boolean

  id: number | string
  name: string
  type: RecipeType
  original_calories?: number
  original_fats?: number
  original_carbs?: number
  original_protein?: number
  calories: number
  fats: number
  carbs: number
  protein: number
  ingredients?: IBaseIngredient[]
  easy_add_recipes?: IEasyAddRecipeIngredient[]
  user_generated?: boolean
  isDirty?: boolean
  thumb_img_url?: string
  img_url?: string
  steps?: IRecipeStep[]
  meal_time_slots?: MealTimeSlot[]
  tools?: IRecipeTool[]
  instructions?: string
  servings?: number
  prep_time?: number
  total_time?: number
  tags?: EatingPreference[]

  food_name?: string
  nix_item_id?: string | null
  brand_name?: string | null
  serving_qty?: number
  serving_unit?: string
  total_fat?: number
  trans_fat?: number | null
  saturated_fat?: number
  cholesterol?: number
  sodium?: number
  total_carbohydrate?: number
  dietary_fiber?: number
  sugars?: number
  potassium?: number
  full_nutrients?: {
    attr_id: number
    value: number
  }[]
  serving_weight_grams?: number
  photo?: {
    thumb: string
    highres: string | null
    is_user_uploaded: boolean
  }
  alt_measures?: AltMeasure[] | null
}

export interface IFavoriteMeal {
  id: number
  name: string
  type: string // recipe | easy_add_recipe | meal_set
  created_at: string
  user_genderated?: boolean
  img_url?: string
  original_recipe_id?: number
}

export interface IMealFilter {
  recipes: {
    favoritesOnly: boolean
    mealTimes: Record<MealTimeSlot, boolean>
    eatingPreferences: Record<Capitalize<Lowercase<EatingPreference>>, boolean>
  }
  mealPlans: {
    favoritesOnly: boolean
    eatingPreferences: Record<Capitalize<Lowercase<EatingPreference>>, boolean>
  }
}

/* ************** STATE ************** */

export interface IMealPlanState {
  day: Date | string
  mealTimeSlot: number
  mealSets: IMealPlan[]
  mealPlans: {
    [key: string]: {
      [key: string]: {
        [key: string]: {
          data: IMealPlanDay
        }
      }
    }
  }
  mealPlansv4: {}
  groceries: GroceryIngredient[]
  selectedGroceries: GroceryIngredient[]
  dates: string
  favorites: IFavoriteMeal[]
  history: Partial<IRecipe>[]
  initialDataLoaded: boolean
  persistedScreenState: Record<string, any>
  currentRecipe: IRecipe | null // TODO still need to figure out what a Recipe is. So many variations
  setOn: Date | string | null
  filters: IMealFilter | null
  isLoading: boolean
  ingredientsOnly: boolean
}

/* ********** GUARDS ********** */
export function isEasyAddRecipeIngredient(ing: any): ing is IEasyAddRecipeIngredient {
  return ing.type === RecipeType.easy_add_recipe
}

/* ************** Api responses and action payloads ******************************* */

export interface IListMealsResponse extends IMealPlanDay {}

export interface IGetRecipesRangePayload {
  startDate: string
  endDate?: string
}

export interface IListMealSetsPayload {
  eatingPreferences: {
    [key in Capitalize<Lowercase<EatingPreference>>]: boolean
  }
  favoritesOnly: boolean
}

export interface IListRecipesResponse {
  data: {
    id: number
    name: string
    thumb_img_url: string
    tags: EatingPreference[]
    meal_time_slots: MealTimeSlot[]
    calories: number
    fats: number
    carbs: number
    protein: number
    is_favorite: boolean
    user_generated: boolean
    original_recipe_id: number | null
    children_ids: number[]
  }[]
  meta: {
    pagination: {
      total: number
      count: number
      per_page: number
      current_page: number
      total_pages: number
      links: Record<string, any>
    }
  }
}

export interface IIngredientsSearchResponse {
  common: {
    food_name: string
    serving_unit: string
    tag_name: string
    serving_qty: number
    common_type: string | null
    tag_id: number
    photo: {
      thumb: string
    }
    locale: string
  }[]
  branded: {
    food_name: string
    serving_unit: string
    nix_brand_id: string
    brand_name_item_name: string
    serving_qty: number
    nf_calories: number
    photo: {
      thumb: string
      highres: string | null
      is_user_uploaded: boolean
    }
    brand_name: string
    region: number
    brand_type: string
    nix_item_id: string
    locale: string
  }[]
}

export interface IRenderableIngredientItem extends Partial<IRecipe> {
  id: number
  type: RecipeType
  key: string
  disableRightSwipe?: boolean
  base_ingredient_id: number
  nix_food_name: string
  meal_time_slot_id: number
  serving_qty: number
  serving_weight_grams: number
  calories: number
  protein: number
}

export interface IIngredientDetailsResponse extends IRenderableIngredientItem {
  error?: boolean
}

export interface IRecipeDetailsResponse {
  id: number
  name: string
  type_id: number
  yields: number
  total_time: number
  mpr_id: number
  instructions: string
  goal_type_id: number
  calories: number
  carbs: number
  fats: number
  protein: number
  prep_time: number
  tag: Capitalize<Lowercase<EatingPreference>>
  full_img_url: string
  thumb_img_url: string
  meal_time_slot?: MealTimeSlot
  meal_time_slot_id?: number
  tags: EatingPreference[]
  ingredients: IMealPlanIngredient[]
  steps: IRecipeStep[]
  tools: IRecipeTool[]
  eating_preferences: EatingPreference[]
}

export interface IMealSetDetailsNewResponse extends IMealPlan {}

export interface IStoreMealSetPayload {
  date: string | Date
  meal_set_id: number
}

export interface IStoreMealSetResponse extends IMealPlanDay {}

export interface IStoreIngredientPayload {
  id: string
  date: string | Date
  meal_time_slot_id: number
  quantity: number
  protein: number
  serving_unit: string
  carbs: number
  fats: number
  calories: number
}

export interface IStoreIngredientResponse extends IMealPlanDay {}

export interface ICreateEasyAddPayload extends IEasyAdd {}

export interface ICreateEasyAddResponse extends IEasyAdd {}

export interface IUpdateEasyAddResponse extends IEasyAdd {}

export interface IEasyAddDetailsResponse extends IEasyAdd {}

export interface IEasyAddSearchResponse extends IEasyAdd {}

export interface IAddEasyAddToDiaryPayload {
  date: string | Date
  easy_add_recipe_id: number
  meal_time_slot_id: number
}

export interface IAddEasyAddToDiaryResponse extends IMealPlanDay {}

export interface IAddRecipeToDiaryPayload {
  date: string | Date
  recipe_id: number | string
  meal_time_slot_id: number
}

export interface IAddRecipeToDiaryResponse extends IMealPlanDay {}

export interface IAddRecipeToFavoritesResponse {
  data: IFavoriteMeal[]
}

export interface IRemoveRecipeFromFavoritesResponse {
  data: IFavoriteMeal[]
}

export interface IAddEasyAddToFavoritesResponse {
  data: IFavoriteMeal[]
}

export interface IRemoveEasyAddFromFavoritesResponse {
  data: IFavoriteMeal[]
}

export interface IAddMealSetToFavoritesResponse {
  data: IFavoriteMeal[]
}

export interface IRemoveMealSetFromFavoritesResponse {
  data: IFavoriteMeal[]
}

export interface IListFavoritesResponse {
  data: IFavoriteMeal[]
}

export type SelectedItem = { type: RecipeType; id: number; meal_time_slot_id: number }
export interface IDeleteDiaryItemPayload {
  date: string | Date
  items: SelectedItem[]
}

export interface IDeleteDiaryItemsResponse {
  data: IMealPlanDay
}

export interface ICopyDiaryItemsPayload {
  dates: string[] | Date[]
  items: {
    id: number
    type: string // recipe | ingredient | easy_add_recipe
    meal_time_slot_id: number
  }[]
}

export interface ICopyDiaryItemsResponse {
  data: IMealPlanDay[]
}

export interface IUpdateIngredientPayload {
  date: string | Date
  data: {
    id: number
    calories: number
    carbs: number
    protein: number
    fats: number
    quantity: number
    serving_unit: string
    meal_time_slot_id: number
  }
}

export interface IUpdateDiaryIngredientResponse {
  data: IMealPlanDay
}

export interface ICreateUserRecipePayload {
  recipe_id: number | string
  name: string
  ingredients: IBaseIngredient[]
  easy_add_recipe_ids: number[] | string[]
}

export interface ICreateUserRecipeResponse {
  data: IIndividualRecipeDetails
}

export interface IUpdateUserRecipePayload {
  id: number | string
  recipe: {
    recipe_id: number | string
    name?: string
    ingredients: IBaseIngredient[]
    easy_add_recipe_ids: number[] | string[]
  }
}
export interface IUpdateUserRecipeResponse {
  data: IIndividualRecipeDetails
}

export interface IStoreMealPlanPayload {
  date: string | Date
  data: IMealPlanDay | null
}

export interface IStoreFavoritesPayload {
  data: IFavoriteMeal[]
}

export interface IStoreItemHistoryPayload {
  item: Partial<IRecipe>
}

export interface IStoreGroceriesPayload {
  dates: string
  ingredients: GroceryIngredient[]
}
