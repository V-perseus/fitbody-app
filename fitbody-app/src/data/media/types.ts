export const FETCH_MEDIA = 'media/fetchMedia'

export enum VideoCategories {
  'Fitness' = 'Fitness',
  'Food' = 'Food',
  'Rehabilitation' = 'Rehabilitation',
  'Body Love' = 'Body Love',
  'Cardio' = 'Cardio',
  'Low Impact' = 'Low Impact',
  'Yoga Flows' = 'Yoga Flows',
  'Postpartum' = 'Postpartum',
  '30 Day Tone' = '30 Day Tone',
}

export enum CategoryCategory {
  'Guidance' = 'Guidance',
  'On Demand' = 'On Demand',
}

export enum GuidanceCategory {
  'Fitness' = 'Fitness',
  'Food' = 'Food',
  'Rehabilitation' = 'Rehabilitation',
  'Body Love' = 'Body Love',
}

export enum OnDemandCategory {
  'Cardio' = 'Cardio',
  'Low Impact' = 'Low Impact',
  'Yoga Flows' = 'Yoga Flows',
  'Postpartum' = 'Postpartum',
  '30 Day Tone' = '30 Day Tone',
}

export interface IVideoData {
  HLS_GROUP: string[]
  DASH_ISO_GROUP?: string[]
  FILE_GROUP?: string[]
}

export interface IVideo {
  id?: number
  title?: string
  video_type_id?: number
  video_category_id?: number
  category_id: number
  trainer_id: number
  duration: number
  thumbnail: string
  body_focus: string | null
  impact: string | null
  intensity: number | null
  video?: string
  video_data: IVideoData
  aws_encode_id?: string
  aws_data?: string
  encoding_complete?: 0 | 1
  created_at?: string
  updated_at?: string
}

export interface IVideoType {
  id?: number
  name: string
  created_at?: string
  updated_at?: string
}

export interface IVideoTypeResponse extends IVideoType {
  categories: IVideoCategory[]
}

export interface IVideoCategory {
  id?: number
  name: string
  icon?: string
  icon_url: string
  image?: string
  image_url: string
  header_image?: string
  header_image_url: string
  offline_image?: string
  offline_image_url: string
  created_at?: string
  updated_at?: string
  video_type_id?: number
  description: string
  class_description: string
  sort_order: number
  hidden: boolean
  period_term: string
  classes_count_override?: number
  session_duration: string
  period_label: string
  short_description?: string
  show_on_quiz: boolean
}

// export type ExcludeName<T> = Pick<T, Exclude<keyof T, 'name'>>;

// export interface ICombinedVideo extends ExcludeName<IVideoCategory>, ExcludeName<IVideoType> {
//   name: VideoCategories | CategoryCategory
// }

export interface IFetchMediaResponse {
  types: IVideoTypeResponse[]
  videos: IVideo[]
}

export interface IMediaVideo {
  Guidance: MediaGrouped[]
  'On Demand': MediaGrouped[]
}

export type MediaGrouped = {
  [key in VideoCategories]: IVideo[]
}

export type GuidanceState = {
  [key in GuidanceCategory]: IVideo[]
}

export type OnDemandState = {
  [key in OnDemandCategory]: IVideo[]
}

export type CategoryState = {
  [key in CategoryCategory]: IVideoCategory[]
}

export interface IDownloadLinks {
  Url: string
  ZipPath: string
  skipDownload?: boolean
}

export interface IDownloadData {
  categories: CategoryState
  guidance: MediaGrouped
  onDemand: MediaGrouped
}

export interface IDownloadState {
  downloadId: string
  progress: number
  payloadToSave: IDownloadData
  links: IDownloadLinks[]
  type: 'media'
  additionalFields?: { [key: string]: any }
}
