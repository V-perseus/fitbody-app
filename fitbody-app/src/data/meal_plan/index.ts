import { store } from '../../store'
import {
  // getRecipesForTheDay,
  getRecipes as gr,
  getRecipesRange as grr,
  listMealSets as lms,
  changeDate as cd,
  changeMealTimeSlot as cmts,
  storeMealPlan as smp,
  storeFavorites as sf,
  storeItemHistory as sih,
  storeFilter as sFil,
  storeGroceries as sg,
  storeSelectedGroceries as ssg,
  deleteGrocery as dg,
  editRecipe as er,
  removeIngredient as ri,
  updateIngredient as ui,
  addIngredient as ai,
  addEasyAddIngredient as aeai,
  persistScreenState as pss,
  removeEasyAddIngredient as reai,
  addMealsetToFavorites as amstf,
  removeMealsetFromFavorites as rmsff,
  setIngredientsOnly as sio,
} from './mealPlanSlice'
import {
  GroceryIngredient,
  IBaseIngredient,
  IEasyAddRecipeIngredient,
  IGetRecipesRangePayload,
  IIndividualRecipeDetails,
  IListMealSetsPayload,
  IMealFilter,
  IStoreFavoritesPayload,
  IStoreGroceriesPayload,
  IStoreItemHistoryPayload,
  IStoreMealPlanPayload,
} from './types'

// export const getRecipesV3 = (data) => {
//   store.dispatch(getRecipesForTheDay(data))
// }

export const getRecipes = (date: string) => {
  store.dispatch(gr(date))
}

export const getRecipesRange = (dates: IGetRecipesRangePayload) => {
  store.dispatch(grr(dates))
}

export const listMealSets = (filter: IListMealSetsPayload) => {
  store.dispatch(lms(filter))
}

export const changeDate = (date: string | Date) => {
  store.dispatch(cd(date))
}

export const changeMealTimeSlot = (data: number) => {
  store.dispatch(cmts({ mealTimeSlotId: data }))
}

export const storeMealPlan = (data: IStoreMealPlanPayload) => {
  store.dispatch(smp(data))
}

export const storeFavorites = (data: IStoreFavoritesPayload) => {
  store.dispatch(sf(data))
}

export const storeItemHistory = (data: IStoreItemHistoryPayload) => {
  store.dispatch(sih(data))
}

export const storeFilter = (data: IMealFilter) => {
  store.dispatch(sFil(data))
}

export const storeGroceries = (data: IStoreGroceriesPayload) => {
  store.dispatch(sg(data))
}

export const storeSelectedGroceries = (data: GroceryIngredient[]) => {
  store.dispatch(ssg(data))
}

export const deleteGrocery = (data: { id: number }) => {
  store.dispatch(dg(data))
}

export const editRecipe = (data: { recipe: IIndividualRecipeDetails }) => {
  store.dispatch(er(data))
}

export const removeIngredient = (data: { id: number }) => {
  store.dispatch(ri(data))
}

export const removeEasyAddIngredient = (data: { id: string }) => {
  store.dispatch(reai(data))
}

export const updateIngredient = (data: { ingredient: IBaseIngredient }) => {
  store.dispatch(ui(data))
}

export const addIngredient = (data: { ingredient: IBaseIngredient }) => {
  store.dispatch(ai(data))
}

export const addEasyAddIngredient = (data: { ingredient: IEasyAddRecipeIngredient }) => {
  store.dispatch(aeai(data))
}

// TODO type this here and in mealPlanSlice
export const persistScreenState = (data: any) => {
  store.dispatch(pss(data))
}

export const addMealsetToFavorites = (id: number) => {
  store.dispatch(amstf(id))
}

export const removeMealsetFromFavorites = (id: number) => {
  store.dispatch(rmsff(id))
}

export const setIngredientsOnly = (val: boolean) => {
  store.dispatch(sio(val))
}
