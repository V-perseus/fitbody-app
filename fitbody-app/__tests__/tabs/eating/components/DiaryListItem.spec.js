import React from 'react'
import { act, cleanup, fireEvent, render, waitFor } from '@testing-library/react-native'
import renderer from 'react-test-renderer'

import DiaryListItem from '../../../../src/screens/tabs/eating/MealPlan/DiaryListItem'
import { RecipeType } from '../../../../src/data/meal_plan/types'
import api from '../../../../src/services/api'

jest.mock('../../../../src/services/api')

describe('DiaryListItem', () => {
  afterEach(() => {
    cleanup()
  })

  let component = null
  beforeEach(async () => {
    // this is called in the initial useEffect
    await api.eating.ingredientDetails.mockResolvedValue({ id: 1, serving_qty: 2, serving_unit: 'cups', name: 'test_ingredient_name' })

    component = (props) => (
      <DiaryListItem
        recipeEdit={true}
        item={{
          id: 1,
          type: props.type || RecipeType.recipe,
          key: 'key',
          disableRightSwipe: false,
          base_ingredient_id: 2,
          nix_food_name: 'nix_food_name',
          meal_time_slot_id: 1,
          serving_qty: 1,
          serving_unit: 'cups',
          serving_weight_grams: 100,
          calories: 200,
          protein: 300,
          name: 'recipe name',
          thumb_img_url: 'test_url',
        }}
        favorites={[{ type: RecipeType.recipe, name: 'recipe name' }]}
        singleEditMode={props.singleEditMode || true}
        onSave={props.onSave}
        onToggleSelect={(items, selectionActive, toggleAll = false) => {}}
        selected={[]}
        editMode={props.editMode || true}
        // date={moment().format('YYYY-MM-DD')}
        slot={{ id: 1 }}
        // handlePress={data.item.type === 'recipe' ? (id) => openRecipe(mealPlan.meal_set.id, id) : addFood}
        handlePress={() => {}}
        ingredientPress={() => {}}
        showMacros={true}
      />
    )
  })
  it('renders snapshot correctly', () => {
    const tree = renderer.create(component({ onSave: jest.fn })).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should render recipe item details', async () => {
    const onSave = jest.fn()
    const { queryByText, queryByTestId } = render(component({ onSave }))
    await act(async () => {
      expect(queryByText(/recipe name/gi)).not.toBeNull()
      expect(queryByTestId('recipe_img')).not.toBeNull()
      expect(queryByText(/FIT BODY RECIPE/gi)).not.toBeNull()
      expect(queryByTestId('recipe_img')).toHaveProp('source', { uri: 'test_url' })
      expect(queryByTestId('edit_recipe_btn')).not.toBeNull()
      await waitFor(() => queryByTestId('favorite_recipe_btn'))
      expect(queryByTestId('favorite_recipe_btn')).not.toBeNull()
    })
  })

  it('should render easy add item details', async () => {
    const onSave = jest.fn()
    const { queryByText, queryByTestId } = render(component({ onSave, type: RecipeType.easy_add_recipe }))
    await act(async () => {
      expect(queryByText(/recipe name/gi)).not.toBeNull()
      expect(queryByText(/EASY ADD/gi)).not.toBeNull()
      expect(queryByText(/200/gi)).not.toBeNull()
      expect(queryByTestId('edit_easy_add_btn')).not.toBeNull()
    })
  })

  it('should render default item details', async () => {
    const onSave = jest.fn()
    const { queryByText, queryByTestId, queryAllByText } = render(component({ onSave, type: RecipeType.ingredient }))
    await act(async () => {
      expect(queryByText(/recipe name/gi)).not.toBeNull()
      expect(queryByText(/Serving Size:/gi)).not.toBeNull()
      expect(queryAllByText(/cup/gi)).toHaveLength(2)
      expect(queryByText(/Number of Servings:/gi)).not.toBeNull()
      expect(queryByTestId('edit_default_edit_btn')).not.toBeNull()
      expect(queryByText('SAVE')).not.toBeNull()
    })
  })

  it('should render/call save button', async () => {
    const onSave = jest.fn()

    const { queryByText } = render(component({ onSave, type: RecipeType.ingredient }))

    await act(async () => {
      await api.eating.updateDiaryIngredient.mockResolvedValueOnce({ id: 1, serving_qty: 3 })

      expect(queryByText('SAVE')).not.toBeNull()
      await fireEvent.press(queryByText('SAVE'))
      expect(onSave).toHaveBeenCalled()
    })
  })
})
