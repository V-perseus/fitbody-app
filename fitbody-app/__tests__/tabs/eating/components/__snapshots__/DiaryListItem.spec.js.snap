// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DiaryListItem renders snapshot correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#ffffff",
    }
  }
>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderBottomColor": "#dedede",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "height": 173.91304347826087,
        "justifyContent": "space-between",
        "opacity": 1,
        "paddingHorizontal": 43.47826086956522,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#ffffff",
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="edit_recipe_btn"
      >
        <circle-outline.svg
          color="#dedede"
          height={43.47826086956522}
          style={
            Object {
              "marginRight": 43.47826086956522,
            }
          }
          width={43.47826086956522}
        />
      </View>
      <Image
        source={
          Object {
            "uri": "test_url",
          }
        }
        style={
          Object {
            "borderRadius": 3.2,
            "height": 67.47826086956522,
            "marginRight": 30.797101449275363,
            "width": 67.47826086956522,
          }
        }
        testID="recipe_img"
      />
      <View
        style={
          Object {
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#000000",
                "fontFamily": "OpenSans",
                "fontSize": 16.869565217391305,
              }
            }
          >
            FIT BODY RECIPE
          </Text>
        </View>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "OpenSans-Bold",
              "fontSize": 19.681159420289855,
            }
          }
        >
          recipe name
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 36.231884057971016,
          "justifyContent": "flex-end",
          "marginLeft": 30.797101449275363,
          "width": 72.46376811594203,
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="favorite_recipe_btn"
      >
        <star.svg
          color="#ffc34c"
          height={16}
          style={
            Object {
              "marginRight": 8,
            }
          }
          width={16}
        />
      </View>
      <right.svg
        color="#333333"
        height={18}
        width={18}
      />
    </View>
  </View>
</View>
`;
